sound/soc/sof/utils.c

```diff
@@ -69,6 +69,50 @@ void sof_mailbox_read(struct snd_sof_dev *sdev, u32 offset,
 }
 EXPORT_SYMBOL(sof_mailbox_read);
 
+void sof_mailbox_get_reply(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_ipc_msg *msg = sdev->msg;
+	struct sof_ipc_reply reply;
+	int ret = 0;
+
+	/*
+	 * Sometimes, there is unexpected reply ipc arriving. The reply
+	 * ipc belongs to none of the ipcs sent from driver.
+	 * In this case, the driver must ignore the ipc.
+	 */
+	if (!msg) {
+		dev_warn(sdev->dev, "unexpected ipc interrupt raised!\n");
+		goto out;
+	}
+
+	/* get the generic reply */
+	snd_sof_dsp_mailbox_read(sdev, sdev->host_box.offset, &reply,
+				 sizeof(reply));
+
+	if (reply.error < 0) {
+		memcpy(msg->reply_data, &reply, sizeof(reply));
+		ret = reply.error;
+	} else if (msg->reply_size > 0) {
+		if (reply.hdr.size > msg->reply_size) {
+			dev_err(sdev->dev,
```

Comment:
It looks like that there are cases when firmware will indicate no error in reply but at the same time the size is set to 0 which by itself should be an error as the `reply.hdr.size` _must_ be at least the `sizeof(reply)`

Reply:
@ranj063, it was a comment for myself to point out a bug or regression the v3 would have introduced (I can not ask for changes on my own PR afaik).
I have addressed it in v4.